sdk/servicebus/service-bus/test/internal/connectionString.spec.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseServiceBusConnectionString } from "../../src/util/connectionStringUtils";
+import chai from "chai";
+
+const assert = chai.assert;
+
+describe("Connection String", () => {
+  const expectedNamespace = "my.servicebus.windows.net";
+  const expectedEndpoint = `sb://${expectedNamespace}`;
+  const expectedEntityPath = "my-entityPath";
+  const expectedSharedAccessSignature = "my-shared-access-signature";
+  const expectedSharedKey = "my-shared-key";
+  const expectedSharedKeyName = "my-shared-key-name";
+  const dummyErrorMsg = "You should not be seeing me";
+
+  it("Extracts Service Bus properties with only Endpoint", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & EntityPath", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};EntityPath=${expectedEntityPath}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+    assert.equal(connectionStringProperties.entityPath, expectedEntityPath);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & SAS", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};SharedAccessSignature=${expectedSharedAccessSignature}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+    assert.equal(connectionStringProperties.sharedAccessSignature, expectedSharedAccessSignature);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & SharedKey", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};SharedAccessKey=${expectedSharedKey};SharedAccessKeyName=${expectedSharedKeyName}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
```

Comment:
Not a change you have to make.

I've been doing assert.deepEqual({ object }, expected) - it has nice output when the assertion fails and you get to see all the fields, not just the one that failed to match. Can be helpful sometimes if you're trying to figure out how badly you "missed" the right answer.